protocol/x/delaymsg/keeper/msg_server.go

```diff
@@ -23,21 +25,41 @@ func (k msgServer) DelayMessage(
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	if !k.HasAuthority(msg.GetAuthority()) {
-		panic(fmt.Errorf(
-			"%v is not recognized as a valid authority for sending delayed messages",
+		k.Logger(ctx).Error(
+			"DelayMessage failed because msg.Authority is not recognized as a valid authority for sending messages",
+			"authority",
 			msg.GetAuthority(),
-		))
+		)
+		return nil, errorsmod.Wrapf(
+			types.ErrInvalidInput,
+			"%v is not recognized as a valid authority for sending messages",
+			msg.GetAuthority(),
+		)
 	}
 
 	sdkMsg, err := msg.GetMessage()
```

Comment:
During manual testing to see if I could provoke an error here for testing I discovered nil messages cause panics, filed audit task: https://linear.app/dydx/issue/CORE-593/duplicate-validatebasic-into-msgserver-delaymessage-so-that-getmessage

Reply:
Can you explain a bit further why it panics? Does it panic [here](https://github.com/dydxprotocol/v4-chain/blob/8b749db88ca08b3514150d5b24ae67651d22073d/protocol/x/delaymsg/types/msg_delay_message.go#L35)?

Reply:
yes, I think so. This issue should be caught by ValidateBasic.